src/action-ros-ci.ts

```diff
@@ -233,22 +263,16 @@ async function run() {
 		);
 
 		// Install ROS dependencies for each distribution being sourced
-		for (let rosDistribution of sourceRosBinaryInstallationList) {
-			// For "latest" builds, rosdep often misses some keys, adding "|| true", to
-			// ignore those failures, as it is often non-critical.
+		if (targetRos1Distro) {
 			await execBashCommand(
-				`DEBIAN_FRONTEND=noninteractive RTI_NC_LICENSE_ACCEPTED=yes rosdep install -r --from-paths src --ignore-src --rosdistro ${rosDistribution} -y || true`,
+				`DEBIAN_FRONTEND=noninteractive RTI_NC_LICENSE_ACCEPTED=yes rosdep install -r --from-paths src --ignore-src --rosdistro ${targetRos1Distro} -y || true`,
 				commandPrefix,
 				options
 			);
 		}
-
-		// If no distribution is being sourced, then install dependencies for the latest release
-		if (!sourceRosBinaryInstallation) {
-			// For "latest" builds, rosdep often misses some keys, adding "|| true", to
-			// ignore those failures, as it is often non-critical.
+		if (targetRos2Distro) {
 			await execBashCommand(
-				`DEBIAN_FRONTEND=noninteractive RTI_NC_LICENSE_ACCEPTED=yes rosdep install -r --from-paths src --ignore-src --rosdistro eloquent -y || true`,
+				`DEBIAN_FRONTEND=noninteractive RTI_NC_LICENSE_ACCEPTED=yes rosdep install -r --from-paths src --ignore-src --rosdistro ${targetRos2Distro} -y || true`,
```

Comment:
We need another line in the command - some builds may not actually have gotten any ROS packages installed due to not having a dependency on it, so `/opt/ros/DISTRO/setup.bash` may not exists (`/opt/ros` doesn't exist in this case)

The below needs to be added to our logic, and we should have a test case to catch this
```
mkdir -p /opt/ros/$DISTRO
touch /opt/ros/$DISTRO/setup.bash
```